nlprule/src/compile/impls.rs

```diff
@@ -145,18 +145,25 @@ impl Tagger {
             .map(|(i, x)| (x.to_string(), PosIdInt::from_value_unchecked(i as u16)))
             .collect();
 
-        let mut tags: WordIdMap<Vec<(WordIdInt, PosIdInt)>> = WordIdMap::new(word_store.len());
+        let mut tags: Vec<Option<Vec<(WordIdInt, PosIdInt)>>> = vec![None; word_store.len()];
```

Comment:
More idiomatic:

`Vec::with_capacity(word_store.len())`

Reply:
That wouldn't work here. We want to treat the vector like a map (manipulate elements up to `word_store.len() - 1` in an arbitrary order) so we actually have to create a vector of this length, not just reserve capacity for it.